examples/gno.land/r/demo/mail/mail.gno

```diff
@@ -0,0 +1,42 @@
+package mail
```

Comment:
What do you think about moving this `Mail` library logic to `examples/gno.land/p/demo/mail`, instead of having it sit in a Realm?

Reply:
Hey Milos!

Thanks taking the time to review this. Yes, deciding what to put in `p/` and `r/` has been one interrogation of mine. 

Here are my thoughts, here is my starting point:

- `p` is package, is like library, is common code for everyone to use, meaning it's quite stable.
- `r` is realm, is app-like, meaning usage is restricted.

By that logic, code that isn't reasonnably finalized at least during GoR is better kept in `r/` as it won't break anything (because nobody will be using it as a dependency). 

I think putting things in `p/demo` is a signal to other devs saying: this code is stable, well-thought out please use it. My app is not there yet, it's WIP.

Another question I have with this app is:

Whether the functionality it offers is really about mails? or about generic messages. 
If it's the latter, then the part to put in `p/` is not mail, it would be like `p/demo/msg`, with `r/demo/mail` being a realm (almost an app instance) using that system. 

This tells me it's a bit early to make the separation now.